packages/router/src/__tests__/nestedSets.test.tsx

```diff
@@ -0,0 +1,199 @@
+            <Route path="/level3" page={Page} name="level3" />
+          </Set>
+        </Set>
+      </Set>
+    </Router>
+  )
+
+  test('level 1, matches expected props', async () => {
+    const screen = render(<TestRouter />)
+
+    act(() => navigate('/level1'))
+
+    await waitFor(() => {
+      expect(screen.queryByText(`Theme: blue`)).toBeInTheDocument()
+      expect(screen.queryByText(`Page Level: 1`)).toBeInTheDocument()
+    })
+  })
+
+  test('level 2, should override level 1', async () => {
+    const screen = render(<TestRouter />)
+
+    act(() => navigate('/level2'))
+
+    await waitFor(() => {
+      expect(screen.queryByText(`Theme: red`)).toBeInTheDocument()
+      expect(screen.queryByText(`Other Prop: bazinga`)).toBeInTheDocument()
+      expect(screen.queryByText(`Page Level: 2`)).toBeInTheDocument()
+    })
+  })
+
+  test('level 3, should override level 1 & 2 and pass through other props', async () => {
+    const screen = render(<TestRouter />)
+
+    act(() => navigate('/level3'))
+
+    await waitFor(() => {
+      expect(screen.queryByText(`Theme: green`)).toBeInTheDocument()
+      expect(screen.queryByText(`Other Prop: bazinga`)).toBeInTheDocument()
+      expect(screen.queryByText(`Page Level: 3`)).toBeInTheDocument()
+    })
+  })
+})
+
+test('Nested sets should not cause a re-mount of wrap components', async () => {
+  const layoutOneMount = jest.fn()
+  const layoutTwoMount = jest.fn()
+
+  const Layout1 = ({ children }) => {
+    useEffect(() => {
+      // Called on mount and re-mount of this layout
+      layoutOneMount()
+    }, [])
+
+    return (
+      <>
+        <p>ONE</p>
+        {children}
+      </>
+    )
+  }
+
+  const Layout2 = ({ children }) => {
+    useEffect(() => {
+      // Called on mount and re-mount of this layout
+      layoutTwoMount()
+    }, [])
+
+    return (
+      <>
+        <p>TWO</p>
+        {children}
+      </>
+    )
+  }
+
+  const NestedSetsWithWrap = () => (
+    <Router>
+      <Set wrap={Layout1}>
+        <Route path="/" page={HomePage} name="home" />
+        <Set wrap={Layout2}>
+          <Route path="/posts/new" page={Page} name="newPost" />
+          <Route path="/posts/{id:Int}/edit" page={Page} name="editPost" />
+          <Route path="/posts/{id:Int}" page={Page} name="post" />
+          <Route path="/posts" page={Page} name="posts" />
+        </Set>
+      </Set>
+    </Router>
+  )
+
+  render(<NestedSetsWithWrap />)
+
+  act(() => navigate('/'))
+  act(() => navigate('/posts'))
+  act(() => navigate('/'))
+  act(() => navigate('/posts'))
+  act(() => navigate('/'))
+
+  // Layout 1 Should only be mounted once, because it's shared between the two sets
+  expect(layoutOneMount).toHaveBeenCalledTimes(1)
+
+  // Layout 2 should be mounted twice, because its used only in the posts routes
+  expect(layoutTwoMount).toHaveBeenCalledTimes(2)
+})
+
+test('Changing the order of wrap components will cause a remount', async () => {
```

Comment:
@KrisCoulson  see this also. Because we're using the index this will happen - and I think that's ok.

If you really wanted to reuse a wrap component what you'd have to do is move it to a parent Set. So

This will remount:
```
      <Set wrap={[Layout1, Layout2]}>
			{..}
			</Set>

	    <Set wrap={[Layout2, Layout3, Layout1]}>
			{..}
			</Set>
```

but, this won't:

```
      <Set wrap={[Layout1, Layout2]}>
			{..}
						<Set wrap={Layout3}>
							{...}
						</Set>
			</Set>
```